pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -344,6 +349,37 @@ func (b *BridgePodInterface) createBridge() error {
 	return nil
 }
 
+func (b *BridgePodInterface) switchPodInterfaceWithDummy() error {
+	originalPodInterfaceName := b.podInterfaceName
+	newPodInterfaceName := fmt.Sprintf("%s-nic", originalPodInterfaceName)
+
+	// Rename pod interface to free the original name for a new dummy interface
+	err := Handler.LinkSetName(b.podNicLink, newPodInterfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to rename interface : %s", b.podInterfaceName)
+		return err
+	}
+	b.podInterfaceName = newPodInterfaceName
+
+	// Create a dummy interface named after the original interface
+	dummy := &netlink.Dummy{LinkAttrs: netlink.LinkAttrs{Name: originalPodInterfaceName}}
```

Comment:
I think there were some concerns on what would happen in cni plugins if we change the interface type. I guess the risk there that something bad is happening, but they could also fail now because of the missing ip. I guess the risk is pretty much the same with this change. We change one property (name) to something unexpected but at the same time ensure that another property is now the same value like before (ip).

Reply:
Let's pretend we are improving it by promoting to the higher level API - L3 instead of L2.